libraries/chain/include/eosio/chain/combined_database.hpp

```diff
@@ -14,77 +36,92 @@
 
 namespace eosio { namespace chain {
 
-   struct combined_session {
+   using controller_index_set =
+         index_set<account_index, account_metadata_index, account_ram_correction_index, global_property_multi_index,
+                   protocol_state_multi_index, dynamic_global_property_multi_index, block_summary_multi_index,
+                   transaction_multi_index, generated_transaction_multi_index, table_id_multi_index, code_index,
+                   database_header_multi_index, kv_db_config_index, kv_index>;
+
+   using contract_database_index_set = index_set<key_value_index, index64_index, index128_index, index256_index,
+                                                 index_double_index, index_long_double_index>;
+
+   class combined_session {
+    public:
+      combined_session() = default;
+
+      combined_session(chainbase::database& cb_database);
+
+      combined_session(chainbase::database& cb_database, b1::chain_kv::undo_stack& kv_undo_stack);
+
+      combined_session(combined_session&& src) noexcept;
+
+      ~combined_session() { undo(); }
+
+      combined_session& operator=(combined_session&& src) noexcept;
```

Comment:
If you didn't add something that needs this, I'd say we should remove it.  I am not sure where we would have a need for an already existing combined_session to have the contents of another combined_session moved into it.  If we don't have it, then we will see if someone does something that needs it, and if it makes sense, we can add it then.

Reply:
I didn't add this operator, and it was used by the `transaction_context` constructor. But because it's a constructor, it was easy to switch from assignment to construction, so I deleted the assignment operator. Sure, we can always add it back when needed.